Packs/AggregatedScripts/Scripts/QuarantineFile/QuarantineFile.py

```diff
@@ -0,0 +1,1145 @@
+            return {}
+
+        # First pass: find all successful, online endpoints. These take precedence.
+        for result in endpoint_data:
+            endpoint_id = result.get("ID")
+            if not endpoint_id or endpoint_id in online_endpoints:
+                continue
+
+            if result.get("Message") == "Command successful" and result.get("Status") == "Online":
+                demisto.debug(f"[EndpointBrandMapper] Found 'Online' status for endpoint {endpoint_id}.")
+                online_endpoints[endpoint_id] = result.get("Brand")
+
+        # Second pass: Create failure results for any endpoint that was found, but not as 'Online'.
+        for result in endpoint_data:
+            endpoint_id = result.get("ID")
+            if not endpoint_id or endpoint_id in all_found_ids:
+                continue
+
+            all_found_ids.add(endpoint_id)
+
+            # Skip endpoints that were found as 'Online'
+            if endpoint_id in online_endpoints:
+                continue
+
+            if result.get("Message") == "Command successful":
+                message = QuarantineResult.Messages.ENDPOINT_STATUS_UNKNOWN.format(status=result.get("Status", "Unknown"))
+            else:  # Message is not 'Command successful', i.e. "Command failed - no endpoint found"
+                message = result.get("Message", QuarantineResult.Messages.ENDPOINT_OFFLINE)
+
+            demisto.debug(f"[EndpointBrandMapper] Creating failure result for endpoint {endpoint_id}. Reason: {message}")
+            self.initial_results.append(
+                QuarantineResult.create(
+                    endpoint_id=endpoint_id,
+                    status=QuarantineResult.Statuses.FAILED,
+                    message=message,
+                    brand=result.get("Brand", "Unknown"),
+                    script_args=self.script_args,
+                )
+            )
+
+        unprocessed_ids = [eid for eid in self.endpoint_ids_to_map if eid not in all_found_ids]
+        if unprocessed_ids:
+            demisto.error(
+                f"[EndpointBrandMapper] Error in get-endpoint-data command. "
+                f"Endpoints not found in any the response: {unprocessed_ids}"
+            )
+            for endpoint_id in unprocessed_ids:
+                self.initial_results.append(
+                    QuarantineResult.create(
+                        endpoint_id=endpoint_id,
+                        status=QuarantineResult.Statuses.FAILED,
+                        message=QuarantineResult.Messages.ENDPOINT_NOT_FOUND,
+                        brand="Unknown",
+                        script_args=self.script_args,
+                    )
+                )
+
+        demisto.debug(f"[EndpointBrandMapper] Processing complete. Found {len(online_endpoints)} online endpoints.")
+        return online_endpoints
+
+
+""" BRAND HANDLER INTERFACE & FACTORY """
+
+
+class BrandHandler:
```

Comment:
i think you can extend ABC in python to make an abstract class - double check the syntax

Reply:
sounds good. 
BTW, this can be an abstract class used as a template for polling commands. 
It has the validate -> pre-process -> kickoff -> finalize logic needed for a polling command.